calc.go

```diff
@@ -1,22 +1,25 @@
 package pokemonbattlelib
 
-func calcMoveDamage(weather Weather, user, receiver *Pokemon, move *Move) (damage uint) {
-	weatherMod := 1.0
-	if rain, sun := weather == WeatherRain, weather == WeatherHarshSunlight; (rain && move.Type() == TypeWater) || (sun && move.Type() == TypeFire) {
-		weatherMod = 1.5
+func calcMoveDamage(b *Battle, user, receiver *Pokemon, move *Move) (damage float64) {
+	weather := 1.0
+	if rain, sun := b.Weather == WeatherRain, b.Weather == WeatherHarshSunlight; (rain && move.Type() == TypeWater) || (sun && move.Type() == TypeFire) {
+		weather = 1.5
 	} else if (rain && move.Type() == TypeFire) || (sun && move.Type() == TypeWater) {
-		weatherMod = 0.5
+		weather = 0.5
+	}
+	crit := 1.0
+	if b.rng.Roll(1, user.CritChance()) {
+		crit = 2.0
```

Comment:
This was intentionally moved outside of this function because it's random. Please put it back where it was.